trio/tests/test_unix_pipes.py

```diff
@@ -98,8 +99,9 @@
     with pytest.raises(TypeError):
         FdStream(None)
 
+    fp = tempfile.TemporaryFile()
     with pytest.raises(ValueError):
-        await FdStream(0).receive_some(0)
+        await FdStream(fp.fileno()).receive_some(0)
```

Comment:
I think if we do it this way, then `TemporaryFile` and `FdStream` will both try to close the same fd from their `__del__` methods? If we're lucky then this will be harmless, but if we're not lucky then we could potentially end up closing an unrelated fd...

How about this:

```python
r, w = os.pipe()
os.close(w)
async with FdStream(r) as s:
    with pytest.raises(ValueError):
        await s.receive_some(0)
```

Reply:
Ah, yes, TemporaryFile has a `__del__` method, it's obvious now that you say it. Thanks, adopted your suggestion!